dotnet/src/SemanticKernel/KernelBuilder.cs

```diff
@@ -40,7 +41,7 @@ public IKernel Build()
         var instance = new Kernel(
             new SkillCollection(this._logger),
             this._aiServices.Build(),
-            new PromptTemplateEngine(this._logger),
+            this._promptTemplateEngine ?? new PromptTemplateEngine(this._logger),
             this._memory,
             this._config,
             this._logger ?? NullLogger<Kernel>.Instance
```

Comment:
The `??` logic for logger only exists for the Kernel, shouldn't all the other be the same?

i.e:
```
var buildLogger = this._logger ?? NullLogger<Kernel>.Instance
var instance = new Kernel(
            new SkillCollection(buildLogger),
            this._aiServices.Build(),
            new PromptTemplateEngine(buildLogger),
            this._memory,
            this._config,
            buildLogger)
```

Reply:
Got you, it appeared that the ?? logic for logger is not needed because the logger field is initialized by default and the WithLogger method does not allow nulls. So, I updated the code to to remove the check.